src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,147 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int EMPTY = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(value, null, null);
+        if (head == null) {
+            head = node;
+        } else {
+            node.prev = tail;
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            Node<T> node = new Node<>(value, head, null);
+            head = node;
+            if (size == 0) {
+                tail = node;
+            }
+            size++;
+        } else if (index == size) {
+            add(value);
+        } else if (validBound(index)) {
+            Node<T> currentNode = getCurrentNode(index);
```

Comment:
This is code repeating. You always check index before getting node by index. What you should do is move the check logic INSIDE the getCurrentNode method. 
![image](https://github.com/mate-academy/jv-linked-list/assets/20326858/117b31cf-cd2d-488e-bf48-78a29ec066ae)